server/controllers/gh.js

```diff
@@ -91,8 +120,56 @@ function makeIssueRequest(issueRequestUrl) {
         });
 }
 
+function makeChangeIssueStateRequest(issueRequestUrl, token, state) {
+    /**
+     * @param issueRequestUrl {String} required, url for API knocking
+     * @param token {String} required, token, generated in Passport to allow user change data
+     * @param state {String} default: 'closed', possible values 'closed' or 'open'
+     * 
+     * @return {Promise} resolve - new state of issue, reject - same error
+     */
+    
+    logger.log('API request to', issueRequestUrl);
+
+    var possibleStates = ['closed', 'open'];
+    state = state || 'closed';
+
+    if (typeof issueRequestUrl !== 'string' || typeof token !== 'string') {
+        return new Promise (function (res, rej) {
+            rej(new Error('Not enough required arguments'));
+        });
+    }
+
+    if (possibleStates.indexOf(state) === -1) {
+        return new Promise(function (res, rej) {
+            rej(new Error('Identifier "state" must be a string with value "closed" or "open"'));
+        });
+    }
+
+    return got(issueRequestUrl, {
+        method: 'PATCH',
+        json: true,
+        headers: {
+            'user-agent': 'bem-forum',
+            accept: 'application/vnd.github.v3.full+json',
+            authorization: 'token ' + token
+        },
+        body: JSON.stringify({
+            state: state
+        })
+    }).then(
+        function (response) {
+            return response.body.state;
+        },
+        function (err) {
```

Comment:
you should better use `.catch` instead and handle the exception (otherwise server will never send respond to the client)